File: internal/usecases/dto/answer.go
package dto

import "strconv"

// =====================
// AnswerForm は回答のフォーム DTO である。
type AnswerForm struct {
	ID        int64       `json:"id"`
	UserID    string      `json:"userId"`
	Content   string      `json:"content"`
	IsFinal   bool        `json:"isFinal"`
	CreatedAt *string     `json:"createdAt,omitempty"`
	UpdatedAt *string     `json:"updatedAt,omitempty"`
	DeletedAt *string     `json:"deletedAt,omitempty"`
	User      *UserForm   `json:"user,omitempty"`
	Refers    []ReferForm `json:"refers,omitempty"`
}

// UserIDInt64 は UserID 文字列を int64 に変換する。
//
// return:
//   - int64: ユーザー ID（変換失敗時は -1）
func (f AnswerForm) UserIDInt64() int64 {
	if val, err := strconv.ParseInt(f.UserID, 10, 64); err == nil {
		return val
	}
	return -1
}

File: static/js/models/Answer.js
/**
 * Answer model class.
 */

import { User } from '/static/js/models/User.js';
import { Refer } from '/static/js/models/Refer.js';
import { formToApiJson } from '/static/js/models/helpers.js';

export class Answer {
  /**
   * @param {number}       id
   * @param {number|string} userId
   * @param {string}       content
   * @param {boolean}      isFinal
   * @param {string}       createdAt   ISO8601 文字列
   * @param {string|null}  updatedAt   ISO8601 文字列
   * @param {User|null}    [user]
   * @param {Refer[]}      [refers]
   */
  constructor(id, userId, content, isFinal, createdAt, updatedAt, user = null, refers = []) {
    this.id = id;
    this.userId = userId;
    this.content = content;
    this.isFinal = Boolean(isFinal);
    this.createdAt = new Date(createdAt);
    this.updatedAt = updatedAt ? new Date(updatedAt) : null;
    this.user = user;
    this.refers = refers;
  }

  /** @param {Object} json @returns {Answer} */
  static fromJSON(json) {
    return new Answer(
      json.id,
      json.userId,
      json.content,
      json.isFinal ?? false,
      json.createdAt ?? new Date().toISOString(),
      json.updatedAt ?? null,
      json.user ? User.fromJSON(json.user) : null,
      (json.refers ?? []).map(Refer.fromJSON),
    );
  }

  /** @returns {Object} */
  static toModel(form) {
    return formToApiJson(form);
  }
}
